### PR TITLE
archive: improve filter error reporting

### PR DESCRIPTION
**Pull request by giuseppe**
follow-up for https://github.com/containers/storage/pull/2012

report the error as seen by the input stream, instead of attempting another read.

Closes: https://github.com/containers/storage/issues/2022

### CODE REVIEW

**Comment by rhatdan** [comment]
LGTM
@mtrmac PTAL

File: pkg/archive/filter.go [hunk 1]
```diff
@@ -26,6 +26,19 @@ func getFilterPath(name string) string {
...
 
+type errorRecordingReader struct {
+	r   io.Reader
+	err error
+}
+
+func (r *errorRecordingReader) Read(p []byte) (int, error) {
+	n, err := r.r.Read(p)
+	if err != nil && err != io.EOF {
+		r.err = err
```
Comment (mtrmac) [inline pkg/archive/filter.go:37]: I think this should prefer the first error, not the last one.

**Review by mtrmac — COMMENTED** [review]
Thanks!

LGTM otherwise.

File: pkg/archive/filter.go (hunk 1 above)
Comment (giuseppe) [reply pkg/archive/filter.go:37]: yes that is better, changed now

Comment (mtrmac) [reply pkg/archive/filter.go:37]: … but still not store `err == nil`?

Comment (mtrmac) [reply pkg/archive/filter.go:37]: … never mind, this works without an explicit check for `err != nil` correctly.

**Pull request merged**